pex/resolve/lock_resolver.py

```diff
@@ -117,9 +117,20 @@ def resolve_from_lock(
         )
         parsed_requirements = try_(
             parse_lockable_requirements(
-                requirement_configuration, network_configuration=network_configuration
+                requirement_configuration,
+                network_configuration=network_configuration,
+                fallback_requirements=lock.requirements,
             )
         )
+        if not parsed_requirements:
```

Comment:
MyPy lying to you again. This will never be None!

Reply:
This wasn't MyPy complaining - this was me being defensive to avoid the `ZeroDivisionError` in case our assertions ever are violated. 

I'll instead take your suggestion on https://github.com/pantsbuild/pex/issues/1659 to tolerate empty lockfiles. So will remove this for now.

Reply:
Moot since it's gone, but you missed the point I think parsed_requirements is a compound object that is never falsey (None) on this line. What you meant to check was `if not parsed_requirements.requirements`. MyPy was truly lying to you since Python supports `__bool__`.